Feature/refactor replicate

xurxodev opened the pull request:
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/869af8g88 #869af8g88
### :memo: Implementation

- [x] Add tests to user entity
- [x] Create username value object and test
- [x] Create password valur oboject and test
- [x] Move generate password to value object
- [x] Manage validation errors using either type
- [x] Move generateUid to user entity

### :video_camera: Screenshots/Screen capture

### :fire: Testing

xurxodev committed:
Avoid static makeErrorMessage be public

xurxodev committed:
Add tests to user entity

xurxodev committed:
Create username value object

xurxodev committed:
Create password value object

xurxodev committed:
Create email value object

xurxodev committed:
Show in the ui the first error

xurxodev committed:
Move generate password to Password value object

ifoche commented:
Task linked: [CU-869af8g88 Refactor to replicate action](https://app.clickup.com/t/869af8g88)

xurxodev committed:
Remove static validateRequiredStringField

xurxodev committed:
Fix severity security

xurxodev committed:
Remove validateRequiredArrayField

xurxodev committed:
Remove static setDefaultLanguage and validate it to create user

xurxodev committed:
Remove unused constant

xurxodev committed:
Remove unnecessary validation because is realised to create by the entity

xurxodev committed:
Move verification about unique openID to use cases:

- because it's a application business rule

xurxodev committed:
Migrate from return error and strings to return either in user

xurxodev committed:
Format the code

xurxodev committed:
Move generateUid to user entity

- The uid generation should be realised by the entity
- create update to replace user properties

eperedo commented on src/domain/value-objects/Password.ts:
```diff
@@ -0,0 +1,120 @@
+            return Either.error([requiredError]);
+        }
+
+        const minLengthError = validateLengthMin(value, 8, "Password should be at least 8 characters long");
```
Probably out of the scope of this PR, but the minLength value depends on a server setting.

<img width="1305" height="342" alt="image" src="https://github.com/user-attachments/assets/ce8ef171-1fe9-4267-8776-9d9151b6a573" />

Maybe we will need a new Entity `PasswordPolicy` or `PasswordRules`? 

We can get this value from the API `/api/systemSettings/minPasswordLength`

https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-240/settings-and-configuration.html (Minimum length of password)

eperedo reviewed (changes requested):
Code looks good! I just have one minor comment about the password minimum length. Also, there are some conflicts with the development branch.

eperedo committed:
Merge branch 'development' of github.com:EyeSeeTea/user-extended-app into feature/refactor_replicate

eperedo committed:
fix and update locales

eperedo commented:
Merge with development @adrianq
